### ISSUE TITLE
[TESTING] Integration Tests Policy Store (Unhappy Path)

### ISSUE THREAD

**Issue by mkanal** [eclipse-tractusx/item-relationship-service#519]
**As** product
**I want** automatically executed integration tests for policy store covering although the unhappy path
**so that** we are informed about regression errors at the earliest stage, and we have a high level of confidence in the application

## Links 
- relates to https://github.com/eclipse-tractusx/item-relationship-service/issues/528

## Acceptance Criteria
- [x] Integration tests (Cucumber or Tavern) covering policy store unhappy path executed in Ci/CD pipeline before PR is merged
- [x] Test covers journey
     - [x] Registration of policy with no validUntil > Should fail 
     - [x] Registration of policy for invalid BPN "BPNACB", "ERRRES", "DELETE * FROM Table" > Should fail 
     - [x] Registration of policy with no payload (poilicy) should fail 
- [x] Registration of BPNA and BPNS number fails 
   - [x] Validation failes for BPNA and BPNS number  

## Out of Scope

- ...

**Comment by dsmf** [comment]
The acceptance criteria "Validation failes for BPNA and BPNS number" contradicts the definition in #505 under Hints, where the valid pattern was defined as "(BPN)[LSA][\w\d]{10}[\w\d]{2}"

**Comment by dsmf** [comment]
also relates to #518

**Comment by dsmf** [comment]
is blocked by #528

**Comment by dsmf** [comment]
outcome:

- [x] Registration of policy with no validUntil > Should fail: https://jira.catena-x.net/browse/TRI-1965
- [x] Registration of policy for invalid BPN "BPNACB", "ERRRES", "DELETE * FROM Table": https://jira.catena-x.net/browse/TRI-1981
- [x] Registration of BPNA and BPNS number fails: https://jira.catena-x.net/browse/TRI-1981
- [x] additionally: Fetching policies by BPNLs should fail for invalid BPNLs: https://jira.catena-x.net/browse/TRI-1966

- [ ] Registration of policy with no payload (poilicy) should fail: TODO

**Comment by dsmf** [comment]
ready for peer review

**Comment by dsmf** [comment]
@mkanal, @ds-kgassner ->  ready for test

**Comment by ds-kgassner** [comment]
Registration of policy with no validUntil > Should fail: --> tested successfully
Registration of policy for invalid BPN "BPNACB", "ERRRES", "DELETE * FROM Table" --> tested successfully
Registration of BPNA and BPNS number fails: --> tested successfully
Registration of policy with no payload (poilicy) should fail: --> tested successfully
additionally: Fetching policies by BPNLs should fail for invalid BPNLs: --> tested successfully

approved from my side

**Comment by dsmf** [comment]
@mkanal -> ready for PO review

**Comment by mkanal** [comment]
LGFM: PO acceptance in behalf of @jzbmw
